controllers/resource/dogu_secret_handler.go

```diff
@@ -0,0 +1,121 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/cesapp/v4/core"
+	"github.com/cloudogu/cesapp/v4/keys"
+	"github.com/cloudogu/cesapp/v4/registry"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+type doguSecretWriter struct {
+	Client   client.Client
+	Registry registry.Registry
+}
+
+// NewDoguSecretsWriter creates a new instance of doguSecretWriter to save key value pairs from secrets to the dogu config
+func NewDoguSecretsWriter(client client.Client, registry registry.Registry) *doguSecretWriter {
+	return &doguSecretWriter{
+		Client:   client,
+		Registry: registry,
+	}
+}
+
+// WriteDoguSecretsToRegistry gets the dogu secret and writes the values in the dogu config. If no secret is found the
+// method returns no error
+func (drr *doguSecretWriter) WriteDoguSecretsToRegistry(ctx context.Context, doguResource *k8sv1.Dogu) error {
+	secret, err := drr.getDoguSetupSecret(ctx, doguResource)
+	if err != nil {
+		return err
+	}
+
+	if secret == nil {
+		return nil
+	}
```

Comment:
We should handle it like the official Kubernetes Go client and return an ErrorNotFound if no config map was found. Then we don't have such dowdy nil checks afterward.